kuflow-temporal-worker/kuflow_temporal_worker/_encryption/_kuflow_encryption_interceptor.py

```diff
@@ -0,0 +1,188 @@
+#
+# MIT License
+#
+# Copyright (c) 2022 KuFlow
+#
+# Permission is hereby granted, free of charge, to any person obtaining a copy
+# of this software and associated documentation files (the "Software"), to deal
+# in the Software without restriction, including without limitation the rights
+# to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+# copies of the Software, and to permit persons to whom the Software is
+# furnished to do so, subject to the following conditions:
+#
+# The above copyright notice and this permission notice shall be included in all
+# copies or substantial portions of the Software.
+#
+# THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+# IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+# FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+# AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+# LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+# OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+# SOFTWARE.
+#
+import dataclasses
+from typing import Any, NoReturn, Optional
+
+import temporalio.client
+import temporalio.worker
+import temporalio.workflow
+
+from ._kuflow_encryption_instrumentation import (
+    KuFlowEncryptionState,
+    KuFlowEncryptionWrapper,
+    add_encryption_encoding,
+    mark_objects_to_be_encrypted,
+    retrieve_encryption_state,
+)
+
+
+class KuFlowEncryptionInterceptor(temporalio.client.Interceptor, temporalio.worker.Interceptor):
+    def intercept_activity(
+        self, next: temporalio.worker.ActivityInboundInterceptor
+    ) -> temporalio.worker.ActivityInboundInterceptor:
+        """Implementation of
+        :py:meth:`temporalio.worker.Interceptor.intercept_activity`.
+        """
+        return KuFlowEncryptionActivityInboundInterceptor(next, self)
+
+    def workflow_interceptor_class(
+        self, input: temporalio.worker.WorkflowInterceptorClassInput
+    ) -> Optional[type[temporalio.worker.WorkflowInboundInterceptor]]:
+        """Implementation of
+        :py:meth:`temporalio.worker.Interceptor.workflow_interceptor_class`.
+        """
+        return KuFlowEncryptionWorkflowInboundInterceptor
+
+
+class KuFlowEncryptionActivityInboundInterceptor(temporalio.worker.ActivityInboundInterceptor):
```

Comment:
Is there no a Activity outbound interceptor?

Reply:
No, this interface intercept request/response invocations to the activities